test/hello_world.test.js

```diff
@@ -1,9 +1,35 @@
 var test = require('tape');
 var HelloWorld = require('../lib/index.js');
+var HW = new HelloWorld();
 
-test('wave hi', function(t) {
-  var HW = new HelloWorld();
+test('wave', function(t) {
   var hello = HW.wave();
-  t.equal(hello, 'howdy world!', 'output of HelloWorld.wave');
+  t.equal(hello, 'howdy world', 'output of HelloWorld.wave');
   t.end();
-})
\ No newline at end of file
+});
+
+test('shout', function(t) {
+  HW.shout('rawr', {}, function(err, shout) {
+    if (err) throw err;
+    console.log(shout);
+    t.end();
+  });
+});
+
+test('shout error - non string phrase', function(t) {
+  HW.shout(4, {}, function(err, shout) {
+    t.ok(err);
```

Comment:
Right, the current code potentially both throws and returns an error in the callback. So the test is incorrect and should do a `try/catch` instead and assert the catch is hit.

This "two ways to catch errors" is not ideal and results in extra code in JS of course. But it is not viable to do the validation in the `HelloWorld::AsyncShout` because one cannot safely copy the data into the threadpool without validating.

Potentially one way to avoid going into the threadpool/still return all errors in the callback would be to call the callback inside `HelloWorld::shout` instead of throwing errors. I think this should work but I've not tried it. Would be a great thing to look into for this repo because figuring out a cleaner way to do this would benefit the usability of all modules based on this code. Can you ticket for a little later?